### ISSUE TITLE
Multiple images upload causes extreme large queue 

### ISSUE THREAD

**Issue by arifje** [nystudio107/craft3-imageoptimize#133]
So, again, let me start with saying I am not sure if this is also caused by Craft itself, or that this plugin triggers something. I am just sharing another issue we've bumped into.

When we upload 50+ images by dragging them, the queue is filling up right away. Not only with "Optimizing image with id X-" taks, but also a lot of "Deleting stale template caches". With 50+ images it took like 30-60 minutes before the queue was done.

The part that I am not getting is why all these "Deleting stale templates caches" tasks are fired. The processing of the 50+ images takes only a short time (big dedicated server, lots of cpu/memory, which we also use for video encoding). But it looks like these tasks are blocked by the cache clearing tasks, which why it is taking ages to complete.

Some screenshots;

![screenshot 2019-01-10 at 16 22 50](https://user-images.githubusercontent.com/1286410/50981101-80651b00-14fa-11e9-90f2-79503711a2b0.png)
![screenshot 2019-01-10 at 16 22 36](https://user-images.githubusercontent.com/1286410/50981102-80fdb180-14fa-11e9-9f62-171ff83be85f.png)

Is there any way to prevent these cache clearing taks to happen or speed up the process? 

Thanks in advance!

**Comment by khalwat** [comment]
So a _long_ queue isn't something you should be worried about; the way Craft queues work, it processes items in the queue one at a time, in a linear fashion.

So you can have as many things in the queue as you like, and it'll just get to them as it gets to them.

What I'd be concerned about is whether or not the queue has been stopped due to errors; you can check `storage/logs/queue.log` and search on `[error]` -- it's possible some really large images are causing PHP to timeout.

At which point you can up the `max_execution_time` and such as needed. Or if you're dealing with really large images very often, you can consider switching over to using Imgix or Thumbor or such, as at some point, there's no getting around that doing large image resizing very often can be CPU intensive.

**Comment by arifje** [comment]
Hi Andrew, thanks for the reply.

Good to know that a long queue isn't something to worry about, but there's still the issue that it takes 30-60 minutes (or sometimes longer if multiple people are uploading articles) before we can publish an article with 50+ images, because of the queue that fills up, which is only happing directly after uploading them. 

And resizing the images isn't the cause in this case; if we manually click on the images, save them (which also triggers the optimising process for that image), it just takes < 2 second per image to create the variations.

So the big question seems to be how we can prevent all these image tasks are blocked by the other processes that are also added to the queue after uploading them. 

I've already tuned my php-cgi configuration to allow more processes/cpu usage. Also changed from GD to Imagick library, but no luck.

**Comment by khalwat** [comment]
So you can't prevent the image tasks from being blocked, that's just how the queue works. But lets see if we can't help you out here regardless.

So the first thing you can try is using the Async Queue plugin: https://github.com/ostark/craft-async-queue which will run your queue via the CLI, and it can have multiple threads running (the default is 4).

Aside from that, can you describe to me *exactly* what you do (the user story) when uploading multiple images? I want to make sure I understand what you're doing on the backend that triggers this, maybe there's something I can discern from it.

**Comment by arifje** [comment]
Ok, so I did some more testing and;

1) disabled GIF conversion in the plugin settings
2) set cacheElementQueries to false in general config

This seems to have solved the issue!
